### ISSUE TITLE
auth.log error messages not captured in fail2ban sshd.conf

### ISSUE THREAD

**Issue by jordanchaitin** [fail2ban/fail2ban#2239]
I use fail2ban with pf on FreeBSD, and there is one kind of log message in auth.log that I do not see in fail2ban's sshd.conf. 

### Environment:
- Fail2Ban version (including any possible distribution suffixes): 0.10.3.fix1
- OS, including release name/version: FreeBSD 11.1-RELEASE-p14
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
I have many sets of messages in my auth.log that look like:
```
Sep 22 13:21:58 xx sshd[26526]: reverse mapping checking getaddrinfo for 78.188.109.175.dynamic.ttnet.com.tr [78.188.109.175] failed - POSSIBLE BREAK-IN ATTEMPT!
Sep 22 13:21:58 xx sshd[26526]: Invalid user user from 78.188.109.175
Sep 22 13:21:58 xx sshd[26526]: input_userauth_request: invalid user user [preauth]
Sep 22 13:21:58 xx sshd[26526]: Connection closed by 78.188.109.175 port 54946 [preauth]
```
I think the key log message in the group above is `Sep 22 13:21:58 xx sshd[26526]: Invalid user user from 78.188.109.175`

Unfortunately, I do not see any rules in fail2ban's sshd.conf that might match the any of the log lines above. In particular, I believe that something like 
`^Invalid user <F-USER> from <HOST> ` 
ought to be added to the list of regexes to match. 

When I do a search for "invalid" in the rules set, I find only these two lines below, neither of which are a match of the log messages I show above. 

```
^Failed publickey for invalid user <F-USER>(?P<cond_user>\S+)|(?:(?! from ).)*?</F-USER> from <HOST>%(__on_port_opt)s(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$
)

^Failed \b(?!publickey)\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>%(__on_port_opt)s(?: ssh\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)

```
On a related note, I would like to add that all the error messages that are used by sshguard should be added to the fail2ban sshd config. The list of error messages of sshguard can be seen [here](https://www.sshguard.net/docs/reference/attack-signatures/)

**Comment by mohag** [comment]
The "reverse mapping" error are [common, mostly for legitimate connections](https://access.redhat.com/solutions/83933) (And you won't get it if you [followed the (not so easy to find) instructions](https://www.fail2ban.org/wiki/index.php/OpenSSH) and set `UseDNS no`) (There are likely another message if the user tries other dodgy things)

The others I agree should be included. (The last one seems to be common in cases that key authentication is the only offered option) (It might make sense to check to ensure that a single mistake does not result in multiple increments of the fail counter though...)

Some more that I have that are not matched: (On Ubuntu 18.04)
```
sshd[30450]: Disconnecting invalid user admin 5.188.10.76 port 39860: Change of username or service not allowed: (admin,ssh-connection) -> (user,ssh-connection) [preauth]
sshd[19320]: Disconnected from authenticating user root 61.184.247.6 port 33553 [preauth]
```

(The last one is my most common log entry for scans currently, on a key-only host, followed by the invalid user message in the OP's report)

Some more: (after sed replacements to allow me to count them (with `sort` and `uniq -x`))
```
Disconnected from invalid user <username> <ip address> port <portnum> [preauth]
Connection closed by authenticating user <username> <ip address> port <portnum> [preauth]
```

(Some might make sense to only include on the aggressive setting)

**Comment by mohag** [comment]
Some samples on how some of these tend to go for a single connection:
```
sshd[27772]: Invalid user hduser from 164.132.101.28 port 34355
sshd[27772]: Received disconnect from 164.132.101.28 port 34355:11: Normal Shutdown, Thank you for playing [preauth]
sshd[27772]: Disconnected from invalid user hduser 164.132.101.28 port 34355 [preauth]
```
No lines matched, at least with the Ubuntu 18.04 version of `sshd.conf` (Matching last line might cause two hits for one mistake)

```
sshd[10401]: Connection closed by authenticating user root 14.187.30.214 port 36321 [preauth]
```
Single line for connection (at least for PID and IP address), not matched.

```
sshd[5672]: Invalid user admin from 5.188.10.76 port 44004
sshd[5672]: Disconnecting invalid user admin 5.188.10.76 port 44004: Change of username or service not allowed: (admin,ssh-connection) -> (user,ssh-connection) [preauth]
```
Two lines, Matching both might make sense - username is invalid, they then try to change it to a different (invalid) user

**Comment by sebres** [comment]
@mohag your fail2ban version? (Ubuntu 18.04 says me nothing).

**Comment by mohag** [comment]
Fail2ban 0.10.2. [Exact level of sshd.conf](https://github.com/fail2ban/fail2ban/blob/2112145eb4a259f1a40a3ce5b709adfa91cb6d5d/config/filter.d/sshd.conf)

My second reply have mostly looked at the version of `sshd.conf` in the repo to see if rules have been added to match them.  (One (as marked) has been added, most of the other changes seems to have been fixes to the rules).

(The main messages that seem to make sense to match as failures):

```
sshd[5672]: Invalid user admin from 5.188.10.76 port 44004
sshd[5672]: Disconnecting invalid user admin 5.188.10.76 port 44004: Change of username or service not allowed: (admin,ssh-connection) -> (user,ssh-connection) [preauth]
```

(The invalid user part (see jordanchaitin's regex) and the "Change of username or service not allowed" part..) (`<USER> <HOST>%(__on_port_opt): Change of username or service not allowed` possibly...) (I'm assuming a similar message might exist for a non invalid username as well. Extracting the username from the intended user ("user" in the example above) instead of the current user might be an option as well... (That one might possibly make sense to limit to preauth as well)

**Comment by sebres** [comment]
closed in #2279
